### PR TITLE
Add onload listener to AB test

### PR DESCRIPTION
**Pull request by buck06191**
## What does this change?

Test if we can use iframe.onload to only run the AB test after the content has loaded.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js [hunk 1]
```diff
@@ -25,18 +25,22 @@ export const newsletterEmbeds: ABTest = {
...
+                            const ifrmElement = (ifrm: HTMLIFrameElement)
+                            ifrmElement.onload = () => {
+                                console.log("THE IFRAME LOADED")
+
+                                const doc = ifrmElement.contentDocument ? ifrm.contentDocument : ifrm.contentWindow.document;
+                                if (doc) {
+                                    const oldDesign = doc.querySelector('.js-ab-embed-old-design');
+                                    const newDesign = doc.querySelector('.js-ab-embed-new-design');
+                                    if (oldDesign && newDesign) {
+                                        oldDesign.classList.add("hide-element")
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js:37]: firing a component event here when we know we're making iframe changes still makes sense to me as a way to check that `variant === new design`

Comment (alexduf) [reply static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js]: done

**Comment by alexduf** [comment]
I've changed the logic slightly to be triggered when the parent frame is loaded, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload):

>The load event fires at the end of the document loading process. At this point, all of the objects in the document are in the DOM, and all the images, scripts, links and sub-frames have finished loading.

Which means we know for sure all the iframes have been loaded

File: static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+        },
+        abTest: {
+            name: 'NewsletterEmbeds',
+            variant
+        }
+    });
+};
 
 export const newsletterEmbeds: ABTest = {
     id: 'NewsletterEmbeds',
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js:19]: Perhaps we should update this ID to make it easy to see when we fixed the issue and restarted the test?

File: static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js [hunk 3]
```diff
@@ -23,25 +38,34 @@ export const newsletterEmbeds: ABTest = {
...
+                        const doc = ifrm.contentDocument ? ifrm.contentDocument : ifrm.contentWindow.document;
+                        window.addEventListener('load', () => {
+                            if (doc) {
+                                const oldDesign = doc.querySelector('.js-ab-embed-old-design');
+                                const newDesign = doc.querySelector('.js-ab-embed-new-design');
+                                if (oldDesign && newDesign) {
+                                    oldDesign.classList.add("hide-element");
+                                    newDesign.classList.remove("hide-element")
+                                }
+                                trackComponentInOphan(ifrm.id, 'variant');
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js:52]: Maybe put this inside the if statement just above it, although `oldDesign` and `newDesign` should always exist once the iframe loads - unless there's some kind of error.

**Review by jfsoul — APPROVED** [review]
Couple of comments that I think are worth doing (and perhaps update the A/B test expiry date as well)

**Pull request merged**